### PR TITLE
create genre uri updated for express-validator v4

### PR DESCRIPTION
**Pull request by VaibhavAjayGupta**
I thought it's cleaner if we move validation and sanitization code into a separate file(requestValidation.js). Please let me know if any drawbacks with requestValidation.js.

### CODE REVIEW

**Comment by hamishwillee** [comment]
Hi @VaibhavAjayGupta 

Thanks for this. I had something almost identical to your code working on Friday. What you have done is how the new module documentation suggests you do things.

Ie, defining validation separately and then calling it as middleware in the router:
```
router.post('/genre/create', validation_controller.genre_create_post_validation, genre_controller.genre_create_post);
```
Thing is, I would much prefer this was all done in the controller - while Express obviously allows you to pass in lots of functions/array of functions there is no "logical" reason to separate the validation logic in that way, and I think it makes things more complicated. Essentially you wouldn't do this if not forced by the API. 

I tried to get this working on Friday  "all in the controller" but haven't yet got it to work. I will try again this Friday.

So I am in two minds about this PR. What do you think "as a user"/reader of our docs. Does using this approach make things harder or is it worth some effort to see if we can get everything in the controller?

**Comment by VaibhavAjayGupta** [comment]
Hi @hamishwillee 
Thank you for your feedback. 
I agree with the point -  not to think about two different places in source to handle a single conceptual operation.
I guess i haven't considered that because validation part and processing part are completely independent operations although must be done synchronously.  
But i still find writing validation code together with processing code is making a single function bulky. 
Will move the code to the controller.

**Comment by hamishwillee** [comment]
I don't find it bulky. I like the fact that you can see (and more importantly describe!) the flow of operations in one place. That said, I haven't yet seen how bulky this will look.
Thanks very much!

**Comment by VaibhavAjayGupta** [comment]
validation code moved back to controller.

File: controllers/genreController.js [hunk 1]
```diff
@@ -46,27 +47,27 @@ exports.genre_create_get = function(req, res, next) {
...
-    req.sanitize('name').escape();
-    req.sanitize('name').trim();
-
-    //Run the validators
-    var errors = req.validationErrors();
+exports.genre_create_post =  [
+    // validate and sanitize request
+   
+    //Check that the name field is not empty also trim and escape the name field.
+      body('name', 'Genre name required').isLength({ min: 1 }).trim().escape(),
```
Comment (hamishwillee) [inline controllers/genreController.js]: Did you test the validation? Specifically, set the min above to 5 and try entering something that requires trimming or escaping. When I tried this last Friday this did not work. The reason was something to with sanitising methods in the validation step only applying during validation. To propagate them you were supposed to use the matcheddata or sanitize. 
Now I may have had other problems with my code affecting this, but best we're sure!

Also, do we need to try/catch some of this?

Comment (VaibhavAjayGupta) [reply controllers/genreController.js]: Thank you for your feedback.
Yep made a mistake shouldn't have combined validation and sanitization. I did ran couple of tests for validation thought jade is modifying the response accordingly on the rendered page. Fixed now. 

I guess we don't need try/catch block as none of the validation/sanitize method is throwing error.

**Comment by hamishwillee** [comment]
Thanks! I will give this a thorough test on Friday, but it looks good on quick inspection.

**Comment by VaibhavAjayGupta** [comment]
Awesome. Thank you!

**Comment by hamishwillee** [comment]
@VaibhavAjayGupta Thanks very much for this. I've reviewed/tested and it makes sense. I'll start the remainder of the update now.

**Pull request merged**